GDM.HW3.ConditionalsMethods/Program.cs

```diff
@@ -0,0 +1,315 @@
+            }
+            if (f < lastArrayIndex)
+            {
+                ReccursiveQuickArraySort(array, f, lastArrayIndex);
+            }
+            return array;
+        }
+
+        //Task 6
+        static int[] PutZeroBetweenMimMaxValueArray(int[] array)
+        {
+            int[] tempArray = (int[])array.Clone();
+            int[] sortedRandomArray = ReccursiveQuickArraySort(tempArray, 0, tempArray.Length - 1);
+            int min = 0;
+            int max = 0;
+
+            for (int i = 0; i < array.Length; i++)
+            {
+                if (array[i] == sortedRandomArray[0])
+                {
+                    min = i;
+                }
+            }
+            for (int i = 0; i < array.Length; i++)
+            {
+                if (array[i] == sortedRandomArray[sortedRandomArray.Length - 1])
+                {
+                    max = i;
+                }
+            }
+            if (min > max)
+            {
+                for (int i = max + 1; i < min; i++)
+                {
+                    array[i] = 0;
+                }
+            }
+            else
+            {
+                for (int i = min + 1; i < max; i++)
+                {
+                    array[i] = 0;
+                }
+            }
+            return array;
+        }
+
+        //Task 8
+        static bool IsDupl(int[] array, int veriable, int arrayIndex)
+        {
+            bool temp = false;
+            for (int i = 0; i < array.Length; i++)
+            {
+                if (array[i] == veriable && arrayIndex != i)
+                {
+                    temp = true;
+                    return temp;
+                }
+            }
+            return temp;
+        }
+        static void Main(string[] args)
+        {
+            Random randomizer = new Random();
+
+            //Task 1
+            string task1 = "Task 1.";
+            var eggsNumberInBox = 10;
+            var eggs = 0;
+            while (eggs < eggsNumberInBox)
+            {
+                eggs++;
+
+            }
+            Console.WriteLine(task1 + $"1. Количество яиц в лотке {eggs} шт.");
+            int eggs2 = 10;
+            int count = 0;
+            do
+            {
+                eggs--;
+                count++;
+
+            } while (eggs != 0);
+            Console.WriteLine(task1 + $"2. При наличии {eggs2} яиц в лотке, можно достать по 1 яйцу {count} раз");
+
+            //Task 2
+            string task2 = "Task 2. ";
+            double fac = randomizer.Next(0, 20);
+            Console.WriteLine(task2 + $"Факториал({fac}) = {Factorial(fac)}");
+
+            //Task 3
+            string task3 = "Task 3. ";
+            int fib = randomizer.Next(0, 30);
+            Console.WriteLine(task3 + $"Последовательностью чисел Фибоначчи для {fib} является:");
+            for (int i = 0; i < fib; i++)
+            {
+                Console.Write(Fibonachi(i) + " ");
```

Comment:
You repeated the same calculation a lot of times. Try to call only one time this function.